mypy/plugins/dataclasses.py

```diff
@@ -931,13 +931,16 @@ def dataclass_tag_callback(ctx: ClassDefContext) -> None:
 
 def dataclass_class_maker_callback(ctx: ClassDefContext) -> bool:
     """Hooks into the class typechecking process to add support for dataclasses."""
-    transformer = DataclassTransformer(
-        ctx.cls, ctx.reason, _get_transform_spec(ctx.reason), ctx.api
-    )
+    spec = _get_transform_spec(ctx.reason)
+    if spec is None:
+        ctx.api.fail("invalid dataclass decorator", ctx.reason, code=errorcodes.MISC)
```

Comment:
Practically we'll never reach this, but wouldn't hurt to alert, since otherwise we'd perform no transform without any other notice.

We can also keep the `assert` we had before, but I feel it's a bit too crash-y.